azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/algorithms/RsaSignature.java

```diff
@@ -10,209 +10,414 @@
+        // emLen - hLen - 1.
+
+        byte[] DB = new byte[emLen - hLen - 1];
+
+        System.arraycopy(PS, 0, DB, 0, emLen - sLen - hLen - 2);
+        DB[emLen - sLen - hLen - 2] = (byte) 0x01;
+        System.arraycopy(salt, 0, DB, emLen - sLen - hLen - 1, sLen);
+
+        // 9. Let dbMask = MGF(H, emLen - hLen - 1).
+        byte[] dbMask = MGF(H, emLen - hLen - 1, messageDigest);
+
+        byte[] maskedDB = new byte[Math.min(DB.length, dbMask.length)];
+
+        // 10. Let maskedDB = DB \xor dbMask.
+        for (int i = 0; i < maskedDB.length; i++) {
+            maskedDB[i] = (byte) (((int) DB[i]) ^ ((int) dbMask[i]));
+        }
+
+        // 11. Set the leftmost 8emLen - emBits bits of the leftmost octet in
+        // maskedDB to zero.
+        int bitMask = 0x000000FF >>> (8 * emLen - emBits);
+        maskedDB[0] &= (byte) bitMask;
+
+        // 12. Let EM = maskedDB || H || 0xbc.
+        return Bytes.concat(maskedDB, H, new byte[] { (byte) 0xbc });
+
     }
     
+    /**
+     * See https://tools.ietf.org/html/rfc3447#section-9.1.2
+     * 
+     * @param mHash the hashed message (hashed with the corresponding digest: i.e., SHA-256 for PS256, SHA-384 for PS384, SHA-512 for PS512
+     * @param EM encoded message, an octet string of emLen = \ceil(emBits/8)
+     * @param emBits maximal bit length of the integer OS2IP(EM)
+     * @param messageDigest the messageDigest that corresponds to the algorithm (see above)
+     * @return true if output consistent with algorithm, false otherwise
+     */
+    protected boolean EMSA_PSS_VERIFY(byte[] mHash, byte[] EM, int emBits, MessageDigest messageDigest) {
+
+        // salt length for the service corresponds to the digest length, which is the hash length / 8
+        int sLen = messageDigest.getDigestLength();
+        int hLen = mHash.length;
+        int emLen = (int) Math.ceil(emBits / 8.0);
+
+        // 3. If emLen < hLen + sLen + 2, output "inconsistent" and stop.
+        if (emLen < hLen + sLen + 2) {
+            return false;
+        }
+
+        // 4. If the rightmost octet of EM does not have hexadecimal value
+        // 0xbc, output "inconsistent" and stop.
+        if (EM[EM.length - 1] != (byte) 0xbc) {
+            return false;
+        }
+
+        // 5. Let maskedDB be the leftmost emLen - hLen - 1 octets of EM, and
+        // let H be the next hLen octets.
+        byte[] maskedDB = Arrays.copyOfRange(EM, 0, emLen - hLen - 1);
+        byte[] H = Arrays.copyOfRange(EM, emLen - hLen - 1, emLen - 1);
+
+        // 6. If the leftmost 8emLen - emBits bits of the leftmost octet in
+        // maskedDB are not all equal to zero, output "inconsistent" and
+        // stop.
+        byte mask = 0x00;
+        for (int i = 0; i < (8 * emLen - emBits); i++) {
+            mask = (byte) (mask | (1 << i));
+            if ((maskedDB[0] & mask) != 0) {
+                return false;
+            }
+            mask = 0x00;
+        }
```

Comment:
Could be simplified to:
byte mask = (byte)0x000000FF >> (8*emLen - emBits);
if(mask | maskedDB[0] != mask) { return false; }